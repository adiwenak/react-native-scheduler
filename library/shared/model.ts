export enum Month {
  January = 0,
  February,
  March,
  April,
  May,
  June,
  July,
  August,
  September,
  October,
  November,
  December
}

export type TimeInMinutes = number
export type CategoryId = string
export type ActivityId = string

// tslint:disable-next-line:max-line-length
export type DateNumber = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 | 21 | 22 | 23 | 24 | 25 | 26 | 27 | 28 | 29 | 30 | 31

export interface BaseTimeline {
  startTime: Date
  endTime: Date
}

export interface Category {
  id: CategoryId
  name: string
  colour: string
}

export interface CategorySelection extends Category {
  isSelected: boolean
}

export interface Activity extends BaseTimeline {
  id: ActivityId
  category: CategoryId
  name: string
}

// CategoryId map to activities
export interface CategoryActivities {
  [key: string]: Activity[]
}

export interface CategoryActivity {
  category: CategorySelection
  tasks: Activity[]
}
