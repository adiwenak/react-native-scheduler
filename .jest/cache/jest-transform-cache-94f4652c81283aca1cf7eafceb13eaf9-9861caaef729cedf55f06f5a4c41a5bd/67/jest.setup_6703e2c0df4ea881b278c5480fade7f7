eabf2f65e7ae4e05e4ba679363e1ad5a
"use strict";var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");var _enzymeAdapterReact = _interopRequireDefault(require("enzyme-adapter-react-16"));
var _enzyme = _interopRequireDefault(require("enzyme"));

_enzyme.default.configure({ adapter: new _enzymeAdapterReact.default() });