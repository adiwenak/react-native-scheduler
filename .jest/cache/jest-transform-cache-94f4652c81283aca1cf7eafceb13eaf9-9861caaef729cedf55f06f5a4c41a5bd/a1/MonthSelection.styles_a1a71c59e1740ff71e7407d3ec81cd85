a13eb5f4de97ed9588efb6ada9bb499b
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styles = void 0;

var _reactNative = require("react-native");

var styles = _reactNative.StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row"
  },
  containerButton: {
    width: 20,
    justifyContent: "center",
    alignItems: "center"
  },
  containerButtonLeft: {
    marginLeft: 17,
    alignItems: "flex-start",
    width: 40
  },
  containerButtonRight: {
    marginRight: 17,
    alignItems: "flex-end",
    width: 40
  },
  containerTitle: {
    flex: 1,
    justifyContent: "center"
  },
  button: {
    width: 20,
    height: 20
  },
  title: {
    textAlign: "center"
  }
});

exports.styles = styles;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi9Vc2Vycy9ha3VybmFkaS9Qcm9qZWN0cy9yZWFjdC1uYXRpdmUtc2NoZWR1bGVyL2xpYnJhcnkvQ2FsZW5kYXJNb250aC9Nb250aFNlbGVjdGlvbi9Nb250aFNlbGVjdGlvbi5zdHlsZXMudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7Ozs7QUFBQTs7QUFZTyxJQUFNLE1BQU0sR0FBRyx3QkFBVyxNQUFYLENBQXlCO0FBQzNDLEVBQUEsU0FBUyxFQUFFO0FBQ1AsSUFBQSxJQUFJLEVBQUUsQ0FEQztBQUVQLElBQUEsYUFBYSxFQUFFO0FBRlIsR0FEZ0M7QUFLM0MsRUFBQSxlQUFlLEVBQUU7QUFDYixJQUFBLEtBQUssRUFBRSxFQURNO0FBRWIsSUFBQSxjQUFjLEVBQUUsUUFGSDtBQUdiLElBQUEsVUFBVSxFQUFFO0FBSEMsR0FMMEI7QUFVM0MsRUFBQSxtQkFBbUIsRUFBRTtBQUNqQixJQUFBLFVBQVUsRUFBRSxFQURLO0FBRWpCLElBQUEsVUFBVSxFQUFFLFlBRks7QUFHakIsSUFBQSxLQUFLLEVBQUU7QUFIVSxHQVZzQjtBQWUzQyxFQUFBLG9CQUFvQixFQUFFO0FBQ2xCLElBQUEsV0FBVyxFQUFFLEVBREs7QUFFbEIsSUFBQSxVQUFVLEVBQUUsVUFGTTtBQUdsQixJQUFBLEtBQUssRUFBRTtBQUhXLEdBZnFCO0FBb0IzQyxFQUFBLGNBQWMsRUFBRTtBQUNaLElBQUEsSUFBSSxFQUFFLENBRE07QUFFWixJQUFBLGNBQWMsRUFBRTtBQUZKLEdBcEIyQjtBQXdCM0MsRUFBQSxNQUFNLEVBQUU7QUFDSixJQUFBLEtBQUssRUFBRSxFQURIO0FBRUosSUFBQSxNQUFNLEVBQUU7QUFGSixHQXhCbUM7QUE0QjNDLEVBQUEsS0FBSyxFQUFFO0FBQ0gsSUFBQSxTQUFTLEVBQUU7QUFEUjtBQTVCb0MsQ0FBekIsQ0FBZiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IEltYWdlU3R5bGUsIFN0eWxlU2hlZXQsIFRleHRTdHlsZSwgVmlld1N0eWxlIH0gZnJvbSBcInJlYWN0LW5hdGl2ZVwiXG5cbmludGVyZmFjZSBTdHlsZSB7XG4gICAgY29udGFpbmVyOiBWaWV3U3R5bGVcbiAgICBjb250YWluZXJCdXR0b246IFZpZXdTdHlsZVxuICAgIGNvbnRhaW5lckJ1dHRvbkxlZnQ6IFZpZXdTdHlsZVxuICAgIGNvbnRhaW5lckJ1dHRvblJpZ2h0OiBWaWV3U3R5bGVcbiAgICBjb250YWluZXJUaXRsZTogVmlld1N0eWxlXG4gICAgdGl0bGU6IFRleHRTdHlsZVxuICAgIGJ1dHRvbjogSW1hZ2VTdHlsZVxufVxuXG5leHBvcnQgY29uc3Qgc3R5bGVzID0gU3R5bGVTaGVldC5jcmVhdGU8U3R5bGU+KHtcbiAgICBjb250YWluZXI6IHtcbiAgICAgICAgZmxleDogMSxcbiAgICAgICAgZmxleERpcmVjdGlvbjogXCJyb3dcIlxuICAgIH0sXG4gICAgY29udGFpbmVyQnV0dG9uOiB7XG4gICAgICAgIHdpZHRoOiAyMCxcbiAgICAgICAganVzdGlmeUNvbnRlbnQ6IFwiY2VudGVyXCIsXG4gICAgICAgIGFsaWduSXRlbXM6IFwiY2VudGVyXCJcbiAgICB9LFxuICAgIGNvbnRhaW5lckJ1dHRvbkxlZnQ6IHtcbiAgICAgICAgbWFyZ2luTGVmdDogMTcsXG4gICAgICAgIGFsaWduSXRlbXM6IFwiZmxleC1zdGFydFwiLFxuICAgICAgICB3aWR0aDogNDBcbiAgICB9LFxuICAgIGNvbnRhaW5lckJ1dHRvblJpZ2h0OiB7XG4gICAgICAgIG1hcmdpblJpZ2h0OiAxNyxcbiAgICAgICAgYWxpZ25JdGVtczogXCJmbGV4LWVuZFwiLFxuICAgICAgICB3aWR0aDogNDBcbiAgICB9LFxuICAgIGNvbnRhaW5lclRpdGxlOiB7XG4gICAgICAgIGZsZXg6IDEsXG4gICAgICAgIGp1c3RpZnlDb250ZW50OiBcImNlbnRlclwiXG4gICAgfSxcbiAgICBidXR0b246IHtcbiAgICAgICAgd2lkdGg6IDIwLFxuICAgICAgICBoZWlnaHQ6IDIwXG4gICAgfSxcbiAgICB0aXRsZToge1xuICAgICAgICB0ZXh0QWxpZ246IFwiY2VudGVyXCJcbiAgICB9XG59KVxuIl19