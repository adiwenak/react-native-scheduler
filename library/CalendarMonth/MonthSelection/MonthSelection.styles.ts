import { ImageStyle, StyleSheet, TextStyle, ViewStyle } from "react-native"

interface Style {
    container: ViewStyle
    containerButton: ViewStyle
    containerButtonLeft: ViewStyle
    containerButtonRight: ViewStyle
    containerTitle: ViewStyle
    title: TextStyle
    button: ImageStyle
}

export const styles = StyleSheet.create<Style>({
    container: {
        flex: 1,
        flexDirection: "row"
    },
    containerButton: {
        width: 20,
        justifyContent: "center",
        alignItems: "center"
    },
    containerButtonLeft: {
        marginLeft: 17,
        alignItems: "flex-start",
        width: 40
    },
    containerButtonRight: {
        marginRight: 17,
        alignItems: "flex-end",
        width: 40
    },
    containerTitle: {
        flex: 1,
        justifyContent: "center"
    },
    button: {
        width: 20,
        height: 20
    },
    title: {
        textAlign: "center"
    }
})
