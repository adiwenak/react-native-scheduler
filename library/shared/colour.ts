// define the name and pick it up from here
// http://chir.ag/projects/name-that-color

export const Colours = {
    // Blue
    Cobalt: "#054BBD",

    // Green
    LaPalma: "#149427",

    // Orange
    Zest: "#E4741B",

    // Red
    Cardinal: "#CC0F20"
}
