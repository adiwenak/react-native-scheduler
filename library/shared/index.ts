export * from "./model"
export * from "./colour"
