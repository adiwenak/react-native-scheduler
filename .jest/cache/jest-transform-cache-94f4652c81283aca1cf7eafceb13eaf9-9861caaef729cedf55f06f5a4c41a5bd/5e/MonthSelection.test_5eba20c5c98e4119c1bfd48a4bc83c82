d56c7542df9db6395cc18ba6bc8207fd
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _enzyme = require("enzyme");

require("jest");

var _react = _interopRequireDefault(require("react"));

require("react-native");

var _reactTestRenderer = require("react-test-renderer");

var _model = require("../../shared/model");

var _MonthSelection = require("./MonthSelection");

describe("<MonthSelection />", function () {
  it("should render with given props", function () {
    var component = (0, _reactTestRenderer.create)(_react.default.createElement(_MonthSelection.MonthSelection, {
      currentYear: 2017
    })).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("should change current month or current year when previous button press", function () {
    var onChange = jest.fn();
    var component = (0, _enzyme.shallow)(_react.default.createElement(_MonthSelection.MonthSelection, {
      currentYear: 2017,
      currentMonth: _model.Month.February,
      onMonthChange: onChange
    }));
    var button = component.find({
      accessibilityLabel: "button-month-prev"
    }).props();
    button.onPress();
    expect(onChange.mock.calls.length).toBe(1);
    expect(component.state("currentMonth")).toBe(_model.Month.January);
    button.onPress();
    expect(onChange.mock.calls.length).toBe(2);
    expect(component.state("currentMonth")).toBe(_model.Month.December);
    expect(component.state("currentYear")).toBe(2016);
  });
  it("should change current month or current year when next button press", function () {
    var onChange = jest.fn();
    var component = (0, _enzyme.shallow)(_react.default.createElement(_MonthSelection.MonthSelection, {
      currentYear: 2017,
      currentMonth: _model.Month.November,
      onMonthChange: onChange
    }));
    var button = component.dive().find({
      accessibilityLabel: "button-month-next"
    }).props();
    button.onPress();
    expect(onChange.mock.calls.length).toBe(1);
    expect(component.state("currentMonth")).toBe(_model.Month.December);
    button.onPress();
    expect(onChange.mock.calls.length).toBe(2);
    expect(component.state("currentMonth")).toBe(_model.Month.January);
    expect(component.state("currentYear")).toBe(2018);
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi9Vc2Vycy9ha3VybmFkaS9Qcm9qZWN0cy9yZWFjdC1uYXRpdmUtc2NoZWR1bGVyL2xpYnJhcnkvQ2FsZW5kYXJNb250aC9Nb250aFNlbGVjdGlvbi9Nb250aFNlbGVjdGlvbi50ZXN0LnRzeCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBOztBQUNBOztBQUNBOztBQUNBOztBQUNBOztBQUNBOztBQUNBOztBQUVBLFFBQVEsQ0FBQyxvQkFBRCxFQUF1QixZQUFBO0FBQzNCLEVBQUEsRUFBRSxDQUFDLGdDQUFELEVBQW1DLFlBQUE7QUFDakMsUUFBTSxTQUFTLEdBQUcsK0JBQ2QsNkJBQUMsOEJBQUQ7QUFBZ0IsTUFBQSxXQUFXLEVBQUU7QUFBN0IsTUFEYyxFQUN3QixNQUR4QixFQUFsQjtBQUVBLElBQUEsTUFBTSxDQUFDLFNBQUQsQ0FBTixDQUFrQixlQUFsQjtBQUNILEdBSkMsQ0FBRjtBQU1BLEVBQUEsRUFBRSxDQUFDLHdFQUFELEVBQTJFLFlBQUE7QUFDekUsUUFBTSxRQUFRLEdBQUcsSUFBSSxDQUFDLEVBQUwsRUFBakI7QUFDQSxRQUFNLFNBQVMsR0FBRyxxQkFDZCw2QkFBQyw4QkFBRDtBQUFnQixNQUFBLFdBQVcsRUFBRSxJQUE3QjtBQUFtQyxNQUFBLFlBQVksRUFBRSxhQUFNLFFBQXZEO0FBQWlFLE1BQUEsYUFBYSxFQUFFO0FBQWhGLE1BRGMsQ0FBbEI7QUFFQSxRQUFNLE1BQU0sR0FBRyxTQUFTLENBQUMsSUFBVixDQUFlO0FBQUUsTUFBQSxrQkFBa0IsRUFBRTtBQUF0QixLQUFmLEVBQTRELEtBQTVELEVBQWY7QUFDQSxJQUFBLE1BQU0sQ0FBQyxPQUFQO0FBQ0EsSUFBQSxNQUFNLENBQUMsUUFBUSxDQUFDLElBQVQsQ0FBYyxLQUFkLENBQW9CLE1BQXJCLENBQU4sQ0FBbUMsSUFBbkMsQ0FBd0MsQ0FBeEM7QUFDQSxJQUFBLE1BQU0sQ0FBQyxTQUFTLENBQUMsS0FBVixDQUFnQixjQUFoQixDQUFELENBQU4sQ0FBd0MsSUFBeEMsQ0FBNkMsYUFBTSxPQUFuRDtBQUVBLElBQUEsTUFBTSxDQUFDLE9BQVA7QUFDQSxJQUFBLE1BQU0sQ0FBQyxRQUFRLENBQUMsSUFBVCxDQUFjLEtBQWQsQ0FBb0IsTUFBckIsQ0FBTixDQUFtQyxJQUFuQyxDQUF3QyxDQUF4QztBQUNBLElBQUEsTUFBTSxDQUFDLFNBQVMsQ0FBQyxLQUFWLENBQWdCLGNBQWhCLENBQUQsQ0FBTixDQUF3QyxJQUF4QyxDQUE2QyxhQUFNLFFBQW5EO0FBQ0EsSUFBQSxNQUFNLENBQUMsU0FBUyxDQUFDLEtBQVYsQ0FBZ0IsYUFBaEIsQ0FBRCxDQUFOLENBQXVDLElBQXZDLENBQTRDLElBQTVDO0FBQ0gsR0FiQyxDQUFGO0FBZUEsRUFBQSxFQUFFLENBQUMsb0VBQUQsRUFBdUUsWUFBQTtBQUNyRSxRQUFNLFFBQVEsR0FBRyxJQUFJLENBQUMsRUFBTCxFQUFqQjtBQUNBLFFBQU0sU0FBUyxHQUFHLHFCQUNkLDZCQUFDLDhCQUFEO0FBQWdCLE1BQUEsV0FBVyxFQUFFLElBQTdCO0FBQW1DLE1BQUEsWUFBWSxFQUFFLGFBQU0sUUFBdkQ7QUFBaUUsTUFBQSxhQUFhLEVBQUU7QUFBaEYsTUFEYyxDQUFsQjtBQUVBLFFBQU0sTUFBTSxHQUFHLFNBQVMsQ0FBQyxJQUFWLEdBQWlCLElBQWpCLENBQXNCO0FBQUUsTUFBQSxrQkFBa0IsRUFBRTtBQUF0QixLQUF0QixFQUFtRSxLQUFuRSxFQUFmO0FBRUEsSUFBQSxNQUFNLENBQUMsT0FBUDtBQUNBLElBQUEsTUFBTSxDQUFDLFFBQVEsQ0FBQyxJQUFULENBQWMsS0FBZCxDQUFvQixNQUFyQixDQUFOLENBQW1DLElBQW5DLENBQXdDLENBQXhDO0FBQ0EsSUFBQSxNQUFNLENBQUMsU0FBUyxDQUFDLEtBQVYsQ0FBZ0IsY0FBaEIsQ0FBRCxDQUFOLENBQXdDLElBQXhDLENBQTZDLGFBQU0sUUFBbkQ7QUFFQSxJQUFBLE1BQU0sQ0FBQyxPQUFQO0FBQ0EsSUFBQSxNQUFNLENBQUMsUUFBUSxDQUFDLElBQVQsQ0FBYyxLQUFkLENBQW9CLE1BQXJCLENBQU4sQ0FBbUMsSUFBbkMsQ0FBd0MsQ0FBeEM7QUFDQSxJQUFBLE1BQU0sQ0FBQyxTQUFTLENBQUMsS0FBVixDQUFnQixjQUFoQixDQUFELENBQU4sQ0FBd0MsSUFBeEMsQ0FBNkMsYUFBTSxPQUFuRDtBQUNBLElBQUEsTUFBTSxDQUFDLFNBQVMsQ0FBQyxLQUFWLENBQWdCLGFBQWhCLENBQUQsQ0FBTixDQUF1QyxJQUF2QyxDQUE0QyxJQUE1QztBQUNILEdBZEMsQ0FBRjtBQWVILENBckNPLENBQVIiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBzaGFsbG93IH0gZnJvbSBcImVuenltZVwiXG5pbXBvcnQgXCJqZXN0XCJcbmltcG9ydCB7IGRlZmF1bHQgYXMgUmVhY3QgfSBmcm9tIFwicmVhY3RcIlxuaW1wb3J0IFwicmVhY3QtbmF0aXZlXCJcbmltcG9ydCB7IGNyZWF0ZSB9IGZyb20gXCJyZWFjdC10ZXN0LXJlbmRlcmVyXCJcbmltcG9ydCB7IE1vbnRoIH0gZnJvbSBcIi4uLy4uL3NoYXJlZC9tb2RlbFwiXG5pbXBvcnQgeyBNb250aFNlbGVjdGlvbiB9IGZyb20gXCIuL01vbnRoU2VsZWN0aW9uXCJcblxuZGVzY3JpYmUoXCI8TW9udGhTZWxlY3Rpb24gLz5cIiwgKCkgPT4ge1xuICAgIGl0KFwic2hvdWxkIHJlbmRlciB3aXRoIGdpdmVuIHByb3BzXCIsICgpID0+IHtcbiAgICAgICAgY29uc3QgY29tcG9uZW50ID0gY3JlYXRlKFxuICAgICAgICAgICAgPE1vbnRoU2VsZWN0aW9uIGN1cnJlbnRZZWFyPXsyMDE3fS8+KS50b0pTT04oKVxuICAgICAgICBleHBlY3QoY29tcG9uZW50KS50b01hdGNoU25hcHNob3QoKVxuICAgIH0pXG5cbiAgICBpdChcInNob3VsZCBjaGFuZ2UgY3VycmVudCBtb250aCBvciBjdXJyZW50IHllYXIgd2hlbiBwcmV2aW91cyBidXR0b24gcHJlc3NcIiwgKCkgPT4ge1xuICAgICAgICBjb25zdCBvbkNoYW5nZSA9IGplc3QuZm4oKVxuICAgICAgICBjb25zdCBjb21wb25lbnQgPSBzaGFsbG93KFxuICAgICAgICAgICAgPE1vbnRoU2VsZWN0aW9uIGN1cnJlbnRZZWFyPXsyMDE3fSBjdXJyZW50TW9udGg9e01vbnRoLkZlYnJ1YXJ5fSBvbk1vbnRoQ2hhbmdlPXtvbkNoYW5nZX0vPilcbiAgICAgICAgY29uc3QgYnV0dG9uID0gY29tcG9uZW50LmZpbmQoeyBhY2Nlc3NpYmlsaXR5TGFiZWw6IFwiYnV0dG9uLW1vbnRoLXByZXZcIiB9KS5wcm9wcygpXG4gICAgICAgIGJ1dHRvbi5vblByZXNzKClcbiAgICAgICAgZXhwZWN0KG9uQ2hhbmdlLm1vY2suY2FsbHMubGVuZ3RoKS50b0JlKDEpXG4gICAgICAgIGV4cGVjdChjb21wb25lbnQuc3RhdGUoXCJjdXJyZW50TW9udGhcIikpLnRvQmUoTW9udGguSmFudWFyeSlcblxuICAgICAgICBidXR0b24ub25QcmVzcygpXG4gICAgICAgIGV4cGVjdChvbkNoYW5nZS5tb2NrLmNhbGxzLmxlbmd0aCkudG9CZSgyKVxuICAgICAgICBleHBlY3QoY29tcG9uZW50LnN0YXRlKFwiY3VycmVudE1vbnRoXCIpKS50b0JlKE1vbnRoLkRlY2VtYmVyKVxuICAgICAgICBleHBlY3QoY29tcG9uZW50LnN0YXRlKFwiY3VycmVudFllYXJcIikpLnRvQmUoMjAxNilcbiAgICB9KVxuXG4gICAgaXQoXCJzaG91bGQgY2hhbmdlIGN1cnJlbnQgbW9udGggb3IgY3VycmVudCB5ZWFyIHdoZW4gbmV4dCBidXR0b24gcHJlc3NcIiwgKCkgPT4ge1xuICAgICAgICBjb25zdCBvbkNoYW5nZSA9IGplc3QuZm4oKVxuICAgICAgICBjb25zdCBjb21wb25lbnQgPSBzaGFsbG93KFxuICAgICAgICAgICAgPE1vbnRoU2VsZWN0aW9uIGN1cnJlbnRZZWFyPXsyMDE3fSBjdXJyZW50TW9udGg9e01vbnRoLk5vdmVtYmVyfSBvbk1vbnRoQ2hhbmdlPXtvbkNoYW5nZX0vPilcbiAgICAgICAgY29uc3QgYnV0dG9uID0gY29tcG9uZW50LmRpdmUoKS5maW5kKHsgYWNjZXNzaWJpbGl0eUxhYmVsOiBcImJ1dHRvbi1tb250aC1uZXh0XCIgfSkucHJvcHMoKVxuXG4gICAgICAgIGJ1dHRvbi5vblByZXNzKClcbiAgICAgICAgZXhwZWN0KG9uQ2hhbmdlLm1vY2suY2FsbHMubGVuZ3RoKS50b0JlKDEpXG4gICAgICAgIGV4cGVjdChjb21wb25lbnQuc3RhdGUoXCJjdXJyZW50TW9udGhcIikpLnRvQmUoTW9udGguRGVjZW1iZXIpXG5cbiAgICAgICAgYnV0dG9uLm9uUHJlc3MoKVxuICAgICAgICBleHBlY3Qob25DaGFuZ2UubW9jay5jYWxscy5sZW5ndGgpLnRvQmUoMilcbiAgICAgICAgZXhwZWN0KGNvbXBvbmVudC5zdGF0ZShcImN1cnJlbnRNb250aFwiKSkudG9CZShNb250aC5KYW51YXJ5KVxuICAgICAgICBleHBlY3QoY29tcG9uZW50LnN0YXRlKFwiY3VycmVudFllYXJcIikpLnRvQmUoMjAxOClcbiAgICB9KVxufSlcbiJdfQ==