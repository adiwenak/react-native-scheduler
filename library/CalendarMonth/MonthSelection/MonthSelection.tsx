import moment from "moment"
import * as React from "react"
import { Component } from "react"
import { Image, Text, TouchableHighlight, View } from "react-native"
import { Month } from "../../shared/model"
import { styles } from "./MonthSelection.styles"

enum MonthAsString {
    January = "January",
    February = "February",
    March = "March",
    April = "April",
    May = "May",
    June = "June",
    July = "July",
    August = "August",
    September = "September",
    October = "October",
    November = "November",
    December = "December"
}

export interface MonthSelectionProps {
    onMonthChange?: (month: Month, year: number) => void
    currentYear?: number
    currentMonth?: Month
}

export interface MonthSelectionState {
    currentMonth: Month
    currentYear: number
}

export class MonthSelection extends Component<MonthSelectionProps, MonthSelectionState> {
    private allMonths: MonthAsString[] = [MonthAsString.January, MonthAsString.February,
        MonthAsString.March, MonthAsString.April, MonthAsString.May,
        MonthAsString.June, MonthAsString.July, MonthAsString.August,
        MonthAsString.September, MonthAsString.October, MonthAsString.November, MonthAsString.December]

    constructor(props: MonthSelectionProps) {
        super(props)
        this.state = {
            currentMonth: props.currentMonth || Month.January,
            currentYear: props.currentYear || moment().year()
        }
    }

    render() {
        const title = `${this.allMonths[this.state.currentMonth]} ${this.state.currentYear}`
        return(
            <View style={styles.container}>
                <TouchableHighlight
                    onPress={this.handlePreviousButtonPress}
                    accessibilityLabel={"button-month-prev"}
                    style={[styles.containerButton, styles.containerButtonLeft]}
                >
                    <Image
                        source={require("./asset/arrow-back.png")} resizeMode={"center"}
                        style={styles.button}
                    />
                </TouchableHighlight>
                <View style={styles.containerTitle}>
                    <Text style={styles.title}>{title}</Text>
                </View>
                <TouchableHighlight
                    onPress={this.handleNextButtonPress}
                    accessibilityLabel={"button-month-next"}
                    style={[styles.containerButton, styles.containerButtonRight]}
                >
                    <Image
                        source={require("./asset/arrow-forward.png")} resizeMode={"center"}
                        style={styles.button}
                    />
                </TouchableHighlight>
            </View >
        )
    }

    private onMonthChange = (month: Month, year: number) => {
        if (this.props.onMonthChange) {
            this.props.onMonthChange(month, year)
        }
    }

    private handlePreviousButtonPress = () => {
        let { currentMonth, currentYear } = this.state
        if (currentMonth > 0) {
            currentMonth -= 1
        } else {
            currentYear -= 1
            currentMonth = Month.December
        }

        this.setState({
            currentMonth,
            currentYear
        })

        this.onMonthChange(currentMonth, currentYear)
    }

    private handleNextButtonPress = () => {
        let { currentMonth, currentYear } = this.state
        if (currentMonth < this.allMonths.length - 1) {
            currentMonth += 1
        } else {
            currentYear += 1
            currentMonth = Month.January
        }

        this.setState({
            currentMonth,
            currentYear
        })

        this.onMonthChange(currentMonth, currentYear)
    }
}
