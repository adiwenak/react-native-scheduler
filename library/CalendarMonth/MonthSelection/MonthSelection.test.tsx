import { shallow } from "enzyme"
import "jest"
import { default as React } from "react"
import "react-native"
import { create } from "react-test-renderer"
import { Month } from "../../shared/model"
import { MonthSelection } from "./MonthSelection"

describe("<MonthSelection />", () => {
    it("should render with given props", () => {
        const component = create(
            <MonthSelection currentYear={2017}/>).toJSON()
        expect(component).toMatchSnapshot()
    })

    it("should change current month or current year when previous button press", () => {
        const onChange = jest.fn()
        const component = shallow(
            <MonthSelection currentYear={2017} currentMonth={Month.February} onMonthChange={onChange}/>)
        const button = component.find({ accessibilityLabel: "button-month-prev" }).props()
        button.onPress()
        expect(onChange.mock.calls.length).toBe(1)
        expect(component.state("currentMonth")).toBe(Month.January)

        button.onPress()
        expect(onChange.mock.calls.length).toBe(2)
        expect(component.state("currentMonth")).toBe(Month.December)
        expect(component.state("currentYear")).toBe(2016)
    })

    it("should change current month or current year when next button press", () => {
        const onChange = jest.fn()
        const component = shallow(
            <MonthSelection currentYear={2017} currentMonth={Month.November} onMonthChange={onChange}/>)
        const button = component.find({ accessibilityLabel: "button-month-next" }).props()

        button.onPress()
        expect(onChange.mock.calls.length).toBe(1)
        expect(component.state("currentMonth")).toBe(Month.December)

        button.onPress()
        expect(onChange.mock.calls.length).toBe(2)
        expect(component.state("currentMonth")).toBe(Month.January)
        expect(component.state("currentYear")).toBe(2018)
    })
})
