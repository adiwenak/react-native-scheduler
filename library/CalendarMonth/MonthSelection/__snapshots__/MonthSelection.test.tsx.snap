// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthSelection /> should render with given props 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel="button-month-prev"
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "alignItems": "flex-start",
          "marginLeft": 17,
          "width": 40,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      resizeMode="center"
      source={
        Object {
          "testUri": "../../../src/MonthView/phone/CalendarMonth/MonthSelection/asset/arrow-back.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      January 2017
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel="button-month-next"
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "alignItems": "flex-end",
          "marginRight": 17,
          "width": 40,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      resizeMode="center"
      source={
        Object {
          "testUri": "../../../src/MonthView/phone/CalendarMonth/MonthSelection/asset/arrow-forward.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
